app/src/main/java/ch/epfl/sweng/SDP/home/HomeActivity.java

```diff
@@ -169,8 +212,30 @@ public boolean onTouch(View view, MotionEvent event) {
         });
     }
 
-    private void listenerEventSelector(final View view, int id) {
-        switch (id) {
+    // Set the listener for the friend request popup buttons using the given userId as the id of the sender of the request
+    private void setFriendsRequestListener(final View view, final double amplitude, final int frequency, final String userId) {
```

Comment:
Since you call this method with mainAmplitude and mainFrequency, remove these parameters and... (see next comment)